python/ray/data/preprocessors/torch.py

```diff
@@ -0,0 +1,94 @@
+        Dataset(num_blocks=..., num_rows=..., schema={image: ArrowTensorType(shape=(..., 3), dtype=float)})
+
+        :class:`TorchVisionPreprocessor` passes ndarrays to your transform. To convert
+        ndarrays to Torch tensors, add ``ToTensor`` to your pipeline.
+
+        >>> from torchvision import transforms
+        >>> from ray.data.preprocessors import TorchVisionPreprocessor
+        >>> transform = transforms.Compose([
+        ...     transforms.ToTensor(),
+        ...     transforms.Resize((224, 224)),
+        ... ])
+        >>> preprocessor = TorchVisionPreprocessor(["image"], transform=transform)
+        >>> preprocessor.transform(dataset)  # doctest: +ellipsis
+        Dataset(num_blocks=..., num_rows=..., schema={image: ArrowTensorType(shape=(3, 224, 224), dtype=float)})
+
+        For better performance, set ``batched`` to ``True`` and replace ``ToTensor``
+        with a batch-supporting ``Lambda``.
+
+        >>> transform = transforms.Compose([
+        ...     transforms.Lambda(
+        ...         lambda batch: torch.as_tensor(batch).permute(0, 3, 1, 2))
+        ...     ),
+        ...     transforms.Resize((224, 224))
+        ... ])
+        >>> preprocessor = TorchVisionPreprocessor(
+        ...     ["image"], transform=transform, batched=True
+        ... )
+        >>> preprocessor.transform(dataset)  # doctest: +ellipsis
+        Dataset(num_blocks=..., num_rows=..., schema={image: ArrowTensorType(shape=(3, 224, 224), dtype=float)})
+
+    Args:
+        columns: The columns to apply the TorchVision transform to.
+        transform: The TorchVision transform you want to apply. This transform should
+            accept an ``np.ndarray`` as input and return a ``torch.Tensor`` as output.
+        batched: If ``True``, apply ``transform`` to batches of shape
+            :math:`(B, H, W, C)`. Otherwise, apply ``transform`` to individual images.
+    """  # noqa: E501
+
+    _is_fittable = False
+
+    def __init__(
+        self,
+        columns: List[str],
+        transform: Callable[["np.ndarray"], "torch.Tensor"],
+        batched: bool = False,
+    ):
+        self._columns = columns
+        self._fn = transform
+        self._batched = batched
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}(columns={self._columns}, "
+            f"transform={self._fn!r})"
+        )
+
+    def _transform_numpy(
+        self, np_data: Union["np.ndarray", Dict[str, "np.ndarray"]]
+    ) -> Union["np.ndarray", Dict[str, "np.ndarray"]]:
+        def transform(batch: np.ndarray) -> np.ndarray:
+            if self._batched:
+                return self._fn(batch).numpy()
+            return np.array([self._fn(array).numpy() for array in batch])
+
+        if isinstance(np_data, dict):
+            outputs = {
+                column: transform(batch)
+                for column, batch in np_data.items()
+                if column in self._columns
+            }
```

Comment:
this currently drops all the columns that are not used by this preprocessor? we should keep those around so that they can continue to be used by downstream preprocessors right?

Reply:
Yeah, good catch. Fixed.